app/src/ui/diff/syntax-highlighting/index.ts

```diff
@@ -74,9 +74,9 @@ async function getOldFileContent(
 async function getNewFileContent(
   repository: Repository,
   file: ChangedFile
-): Promise<Buffer> {
+): Promise<Buffer | null> {
   if (file.status.kind === AppFileStatusKind.Deleted) {
-    return Buffer.alloc(0)
```

Comment:
Because of this we would get an empty string even when the file was deleted, which counted as a 1-line file that messed up with other calculations and made the app think the file could be expanded 😅